### ISSUE TITLE
Override Accept-Language header with en-US

### ISSUE THREAD

**Issue by xxjoe2** [bromite/bromite#2160]
### Preliminary checklist

- [X] I have read the [README](https://github.com/bromite/bromite/blob/master/README.md)
- [X] I have read the [FAQs](https://github.com/bromite/bromite/blob/master/FAQ.md).
- [X] I have searched [existing issues](https://github.com/bromite/bromite/issues) for my feature request. This is a new issue (NOT a duplicate) and is not related to another issue.

### Is your feature request related to privacy?

Yes

### Is there a patch available for this feature somewhere?

couldnt find one for chrome but it should be under class AcceptLanguageBuilder in net/http/http_util.cc

note that changes here MAY not affect language detection with javascript

### Describe the solution you would like

by default, browser will send a header to specify the lang preference in accept-language based on the system language. however, this increase the chance of being identified if users are not us based

by providing an option to override and spoof the accept-language to en-us, fingerprints uniqueness and privacy can be enhanced greatly as shown below

![Screenshot_20220630-163849_Mull](https://user-images.githubusercontent.com/83495134/176634427-3f401c7d-6a04-4bbd-8b9e-3b4bc4bf4078.jpg)
![Screenshot_20220630-163835_Mull](https://user-images.githubusercontent.com/83495134/176634447-e6ba3a04-f208-47ff-90b5-e501ea17c0ed.jpg)

### Describe alternatives you have considered

since we cannot install extension to cope with this we have to hard coded it in the sources

also, since some users are not english or prefer other languages, make it an option to override the value instead of forcing it as some users reported this will change the language shown in some web sites to english

**Comment by uazo** [comment]
related [1323776](https://bugs.chromium.org/p/chromium/issues/detail?id=1323776)

**Comment by csagan5** [comment]
@uazo is there something from upstream we can already use? Otherwise this would look a lot like the UA and timezone spoofing situation.

**Comment by uazo** [comment]
> is there something from upstream we can already use? 

yes, I had checked.
their [implementation](https://bugs.chromium.org/p/chromium/issues/detail?id=1323776) is fine, but it only handles the (http) response and not the request.

we should decide what to do: always set the English language? change the value "q"? follow the timezone override? 
I am of the idea not to send it at all and add a site setting, I'm waiting for your idea

**Comment by csagan5** [comment]
> yes, I had checked.
> their [implementation](https://bugs.chromium.org/p/chromium/issues/detail?id=1323776) is fine, but it only handles the (http) response and not the request.

What do you mean? `Accept-Language` is a HTTP request header, not a response one.



> we should decide what to do: always set the English language? change the value "q"? follow the timezone override?
> I am of the idea not to send it at all and add a site setting, I'm waiting for your idea

Ideally it should follow the timezone, although for each timezone there are tens of possible locales, and only a few have a big user population using them. So this could be a combo box in the timezone settings page. And the list of locales has the most-common ones for that timezone on top.

Is all of this complexity justified for the privacy benefits achieved? No.
A simpler implementation is to let user choose between system-provided (current one) and manually-entered locale, with default of English.

**Comment by uazo** [comment]
> What do you mean? Accept-Language is a HTTP request header, not a response one.

the implementation is of this
```
Implement the Variants and Content-Language response header as a URL
Loader throttle. This restarts/internally redirects the request when it
sees a better user's accept-language available during language
negotiation. For details example: see design docs.
```
ref https://github.com/Tanych/accept-language

> A simpler implementation is to let user choose between system-provided (current one) and manually-entered locale, with default of English.

regarding the latter, I would rather not send it at all, leaving the server to decide how to behave accordingly, which is perhaps better than lying. in my case, for local sites, a server without indications is more likely to respond with content in Italian rather than in English, while for international ones in English or perhaps based on the calling IP
It is true that we could increase the browser's entropy, but I think there are already other methods for identifying bromite as a browser

**Comment by csagan5** [comment]
> what do you think about it?

Sounds good.

**Comment by csagan5** [comment]
Fixed in `105.0.5195.147`.
